### PR TITLE
[EX-5272] Add more smoke tests

### PR DESCRIPTION
**Pull request by mavmaf**
## Motivation and context
After adding first round of smoke test to the archetype, this second one includes:

- Infinite scroll
- URL Navigation
- Multiple consecutive searches.
- Tagging

Since Extra params and Tracking E2E tests are not ready in the X-Components project yet, they were not added now and will be once they are available in X-Components project.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [x] `Main`
- [ ] Other. Specify:

## How has this been tested?

In orden to run the tests independently, please change `cypress.json` `testFiles` parameter to the following:
`"testFiles": "**/*.feature",`

Tests performed according to [testing guidelines](https://github.com/empathyco/x/blob/main/.github/contributing/tests.md): 

## Checklist:

- [x] My code follows the **[style guidelines](https://github.com/empathyco/x/blob/main/.github/CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [ ] I have **commented** my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate **no new warnings**.
- [x] I have added **tests** that prove my fix is effective or that my feature works.
- [x] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

File: tests/e2e/cucumber/multiple-interactions.spec.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    .each($resultTitle => {
+      newResultsList.push($resultTitle.text());
+    });
+});
+
+Then('new related results are different from previous ones', () => {
+  console.log(resultsList[0]);
+  console.log(newResultsList[0]);
+  expect(newResultsList.every(item => resultsList.includes(item))).to.eq(false);
+  resultsList = Object.assign([], newResultsList);
```
Comment (diegopf) [inline tests/e2e/cucumber/multiple-interactions.spec.ts:31]: do we need this? doing `resultList = newResultList` should be enough.

Also, do we want to keep the console.log ?

**Review by diegopf — CHANGES_REQUESTED** [review]

File: tests/e2e/cucumber/scroll.spec.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    });
+});
+
+Then('scroll position is at top', () => {
+  cy.get('#main-scroll').should(scrollContainer => {
+    expect(scrollContainer.scrollTop()).to.equal(0);
+  });
+});
+
+When('store is changed to {string}', (store: string) => {
```
Comment (tiborux) [inline tests/e2e/cucumber/scroll.spec.ts:33]: This is not used

File: tests/e2e/cucumber/tagging.feature [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  Scenario Outline: 1. Check some actions trigger the query tagging.
+    Given start page with "<view>" size view
+    When  start button is clicked
+    And   "<query>" is searched
+    Then  query tagging request should be triggered
+    Given a URL with query parameter "<query>" and "<view>" size view
+    Then  query tagging request should be triggered
+    And   results page number 1 is loaded
+    When  scrolls down to next page
+    Then  results page number <pageNumber> is loaded
```
Comment (tiborux) [inline tests/e2e/cucumber/tagging.feature:16]:
```suggestion
    Then  results page number "<pageNumber>" is loaded
```

File: tests/e2e/cucumber/tagging.feature [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    And   query tagging request has been triggered
+    And   page <pageNumber> query tagging request is triggered
```
Comment (tiborux) [inline tests/e2e/cucumber/tagging.feature:18]:
```suggestion
    And   page "<pageNumber>" query tagging request is triggered
```

**Review by tiborux — CHANGES_REQUESTED** [review]
gj

File: tests/e2e/cucumber/mocked-responses.spec.ts [hunk 5]
```diff
@@ -13,3 +23,18 @@ Given('a results API with a promoted', () => {
 Given('a results API with a banner', () => {
   cy.intercept(searchEndpoint, { fixture: 'search-response-with-banner' }).as('interceptedBanner');
 });
+
+Given('a results API with known results', () => {
+  cy.intercept(searchEndpoint, { fixture: 'search-response-with-known-results' }).as(
+    'interceptedResults'
```
[the remark this answers is not in the document]
Comment (mavmaf) [reply tests/e2e/cucumber/mocked-responses.spec.ts:29]: Yep, as long as there is only one of those `intercepts` per tests. That allows us to use the same `wait `step for all the steps with the same` intercept` alias and this way no code is duplicated.

File: tests/e2e/cucumber/scroll.spec.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import { Then, When } from 'cypress-cucumber-preprocessor/steps';
+
+When('scrolling down to result {string}', (resultId: string) => {
+  cy.get(`[data-scroll=${resultId}]`)
+    .scrollIntoView({ easing: 'swing', offset: { top: -5, left: 0 } })
```
[the remark this answers is not in the document]
Comment (mavmaf) [reply tests/e2e/cucumber/scroll.spec.ts:5]: The offset specifies the amount to scroll after the element (`[data-scroll=${resultId}]`) has been scrolled into view. What I am trying to do here is to ensure that the scroll parameter passed to the URL matches in fact with that element being visible.

File: tests/e2e/cucumber/tagging.feature (hunk 3 above)
Comment (mavmaf) [reply tests/e2e/cucumber/tagging.feature:16]: That's not correct, strings and integers are represented in a different way.

File: tests/e2e/cucumber/tagging.feature (hunk 4 above)
Comment (mavmaf) [reply tests/e2e/cucumber/tagging.feature:18]: That's not correct, strings and integers are represented in a different way.

**Review by diegopf — APPROVED** [review]

**Review by tiborux — APPROVED** [review]

**Pull request merged**